metaserver/auditrunner.go

```diff
@@ -0,0 +1,160 @@
+package metaserver
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// TODO: Refactor these into their own package inside of the provider.
+// Currently, we can't use them from the provider package because it causes
+// an import cycle.
+
+type postAuditParams struct {
+	Nonce string `json:"nonce"`
+}
+
+type postAuditResp struct {
+	Hash string `json:"hash"`
+}
+
+func auditBlock(addr, renterID, blockID, nonce string) (hash string, err error) {
+	client := http.Client{}
+	url := fmt.Sprintf("http://%s/blocks/audit?renterID=%s&blockID=%s", addr, renterID, blockID)
+	req := postAuditParams{nonce}
+	body, err := json.Marshal(&req)
+	if err != nil {
+		return "", err
+	}
+	resp, err := client.Post(url, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", decodeError(resp.Body)
+	}
+	var respMsg postAuditResp
+	err = json.NewDecoder(resp.Body).Decode(&respMsg)
+	if err != nil {
+		return "", err
+	}
+	return respMsg.Hash, nil
+}
+
+func (server *MetaServer) startAuditRunner() {
+	// Frequency at which the runner should be triggered (should probably put this in a config file)
+	runnerFrequency := time.Minute * 5
+
+	// Ticker triggering the runner.
+	ticker := time.NewTicker(runnerFrequency)
+
+	go func() {
+		for range ticker.C {
+			server.logger.Println("Running audits...")
+			err := server.runAudits()
+			if err != nil {
+				server.logger.Println("Error when running audits:", err)
+			}
+		}
+	}()
+}
+
+func (server *MetaServer) runAudits() error {
+	// Retrieve a list of all files on the server.
+	files, err := server.db.FindAllFiles()
```

Comment:
Can we maybe just audit a subset? All files every 5 minutes is quite a bit of activity.

Reply:
That's fine. I figured we would just decrease the frequency in "production" mode (perhaps running it once a day, week, or longer).